opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -311,36 +311,35 @@ namespace Opm {
     template<typename TypeTag>
     void
     BlackoilWellModel<TypeTag>::wellTesting(const int timeStepIdx, const double simulationTime) {
+        Opm::DeferredLogger local_deferredLogger;
         const auto& wtest_config = schedule().wtestConfig(timeStepIdx);
-        if (wtest_config.size() == 0) { // there is no WTEST request
```

Comment:
Did we deadlock previously or what it is this reason for this change?

Reply:
If I understand correctly, than, before this change, logging happened only from rank 0. Now we use if(wtest_config.size() != 0) to test whether we need to execute well testing, and run gatherDeferredLogger afterwards in any case.

Reply:
I see. Makes sense.